packages/playwright-test/types/test.d.ts

```diff
@@ -1310,12 +1319,28 @@ export interface TestInfo {
    *   // Capture a screenshot and attach it.
    *   const path = testInfo.outputPath('screenshot.png');
    *   await page.screenshot({ path });
-   *   testInfo.attachments.push({ name: 'screenshot', path, contentType: 'image/png' });
+   *   await testInfo.attach({ name: 'screenshot', path, contentType: 'image/png' });
    * });
    * ```
    *
+   * Or you can attach files returned by your APIs:
+   *
+   * ```ts
+   * import { test, expect } from '@playwright/test';
+   *
+   * test('basic test', async ({}, testInfo) => {
+   *   const { download } = require('./my-custom-helpers');
+   *   const tmpPath = await download('a');
+   *   await testInfo.attach({ name: 'example.json', path: tmpPath, contentType: 'application/json' });
+   * });
+   * ```
+   *
+   * > NOTE: [testInfo.attach(type:)](https://playwright.dev/docs/api/class-testinfo#test-info-attach) automatically takes
+   * care of copying attachments to a location that is accessible to reporters, even if you were to delete the attachment
+   * after awaiting the attach call.
+   * @param type:
```

Comment:
@dgozman This doesn't look right! I tried to debug—albeit very briefly—and could not resolve. Can you give me some pointers? Maybe my syntax in docs/src/test-api/class-testinfo.md is incorrect. (I did inline the type compared to the last changeset where I had pulled it out to a TestAttachment type.)

Reply:
I made a suggestion in the md file.